examples/financial_document_analysis_with_llamaindex.ipynb

```diff
@@ -0,0 +1,554 @@
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "9aebd935-07e0-44bb-a682-cf1d7b2a9b07",
+   "metadata": {
+    "tags": []
+   },
+   "source": [
+    "In this example notebook, we showcase how to perform financial analysis over [**10-K**](https://en.wikipedia.org/wiki/Form_10-K) documents with the [**LlamaIndex**](https://gpt-index.readthedocs.io/en/latest/) framework with just a few lines of code."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "cb6f657b-b692-44ed-9b8f-2b277e07828d",
+   "metadata": {
+    "tags": []
+   },
+   "source": [
+    "## Notebook Outline\n",
+    "* [Introduction](#Introduction)\n",
+    "* [Setup](#Setup)\n",
+    "* [Data Loading & Indexing](#Data-Loading-and-Indexing)\n",
+    "* [Simple QA](#Simple-QA)\n",
+    "* [Advanced QA - Compare and Contrast](#Advanced-QA---Compare-and-Contrast)\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1e38922b-10ad-4442-bff3-32ee2a384c4d",
+   "metadata": {
+    "tags": []
+   },
+   "source": [
+    "## Introduction\n",
+    "\n",
+    "### LLamaIndex\n",
+    "[LlamaIndex](https://gpt-index.readthedocs.io/en/latest/) is a data framework for LLM applications.\n",
+    "You can get started with just a few lines of code and build a retrieval-augmented generation (RAG) system in minutes.\n",
+    "For more advanced users, LlamaIndex offers a rich toolkit for ingesting and indexing your data, modules for retrieval and re-ranking, and composable components for building custom query engines.\n",
+    "\n",
+    "See [full documentation](https://gpt-index.readthedocs.io/en/latest/) for more details.\n",
+    "\n",
+    "### Financial Analysis over 10-K documents\n",
+    "A key part of a financial analyst's job is to extract information and synthesize insight from long financial documents.\n",
+    "A great example is the 10-K form - an annual report required by the U.S. Securities and Exchange Commission (SEC), that gives a comprehensive summary of a company's financial performance.\n",
+    "These documents typically run hundred of pages in length, and contain domain-specific terminology that makes it challenging for a layperson to digest quickly.\n",
+    "\n",
+    "\n",
+    "We showcase how LlamaIndex can support a financial analyst in quickly extracting information and synthesize insights **across multiple documents** with very little coding. "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "ad72f9c1-e35c-476d-94d8-ca9619d0cc08",
+   "metadata": {
+    "tags": []
+   },
+   "source": [
+    "## Setup"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "15284a54-35b3-4129-9a7a-57b7547bfaf4",
+   "metadata": {
+    "tags": []
+   },
+   "source": [
+    "To begin, we need to install the llama-index library"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "d1062e24-6177-459a-94e0-8e77b3e0859b",
+   "metadata": {
+    "tags": []
+   },
+   "outputs": [],
+   "source": [
+    "!pip install llama-index"
```

Comment:
Can you add pypdf and Langchain here - I got an error on the SimpleDirectoryReader because of no pypdf

Reply:
good catch. will do!

Reply:
Added `pypdf`.

`langchain` is a dependency of `llama-index`, so I think it's unnecessary here.